### PR TITLE
Add details popup for mobile and desktop screens

### PR DESCRIPTION
**Pull request by MarsIncarnate**
This PR meets the following requirements:
- When the user clicks (or taps) the button to check project details, the popup with details about the project appears.
- When the user clicks (or taps) the close (X) button, the popup disappears.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -163,10 +163,33 @@ <h3>Multi-Post Stories<br>Gain+Glory</h3>
...
     </section>
+    <div id="portfolio-details">
+        <img id="detailImg" src="" alt="">
+        <h1 id="detailH1"></h1>
+        <ul id="detailUl"></ul>
+        <p id="detailP"></p>
+        <div class="buttons">
+            <button>
+                See Live
+                <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (TadesseDev) [inline index.html]: - [ ] Good job on the popup, Yet compared with the main design one thing missing is unlike the Figma design the background content is clearly visible on your design.

## Figma design 
Screenshot 2023-01-07 193152
## Your design.
Screenshot 2023-01-07 193228

- [ ] Also, on a screen size around 770px to 1000px your popup seems to be overflowing, Pleas address this for a better responsive design. 

Screenshot 2023-01-07 193729

File: index.js [hunk 2]
```diff
@@ -1,17 +1,90 @@
...
+    p: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releaLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the relea",
+  },
+  {
+    img: 'images/details-image1.png',
+    h1: 'Keeping track of hundreds of components',
+    ul: ['Ruby on rails', 'css', 'JavaScript'],
+    ul2: ['Codekit', 'GitHub', 'JavaScript', 'Bootstrap', 'Terminal'],
+    p: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releaLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the relea",
+  },
```
Comment (TadesseDev) [inline index.js]: - [ ] Dear @MarsIncarnate, I see you have an array of objects and that is good, But Its repetitive content and is not according to the project detail. Each object should contain information that could be used to render a dynamic project that has `name`,  `description`, `image`, `technologies`, `link`

> Something like this
```javascript
const projects = [{
    name: "Submarine game",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
}]
```
> Thanks to these changes adding projects in the future will be easy. You will only add it in one place.

File: index.html [hunk 3]
```diff
@@ -111,7 +111,7 @@ <h3>Multi-Post Stories<br>Gain+Glory</h3>
                     <li>JavaScript</li>
                     <li>html</li>
                 </ul>
-                <button onclick="location.href='#';">See Project</button>
+                <button onclick="revealDetails(1)">See Project</button>
```
Comment (TadesseDev) [inline index.html:104]: - [ ] Dear @MarsIncarnate, according to the main requirement, You need to remove static contents from your HTML and use the array of objects(projects) to render your projects dynamically. `The projects section is created dynamically with the information from the projects array`

##### we need this content to be rendered dynamically from JavaScript.
https://github.com/MarsIncarnate/Microverse_Portfolio/blob/74a324a2c550ea7acad396ae69b94d3c3ef239ca/index.html#L91-L104

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @MarsIncarnate 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost-done-ashley-garcia



## Highlights
 - [ ] Good job overall following the template design ✅
 - [ ]  Correct GitHub Flow  ✅
 - [ ]  Good commit history  ✅
 - [ ]  Good PR title/description  ✅
 - [ ]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html [hunk 4]
```diff
@@ -91,82 +91,77 @@ <h2>My Recent Works
...
+        <div id="portfolio-details">
+            <img id="detailImg" src="" alt="">
+            <h1 id="detailH1"></h1>
+            <ul id="detailUl"></ul>
+            <p id="detailP"></p>
+            <div class="buttons">
+                <button>
+                    See Live
+                    <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+                        <path fill-rule="evenodd" clip-rule="evenodd" d="M5 12C5 8.13401 8.13401 5 12 5C12.5523 5 13 4.55228 13 4C13 3.44772 12.5523 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21C16.9706 21 21 16.9706 21 12C21 11.4477 20.5523 11 20 11C19.4477 11 19 11.4477 19 12C19 15.866 15.866 19 12 19C8.13401 19 5 15.866 5 12ZM16 3C15.4477 3 15 3.44772 15 4C15 4.55228 15.4477 5 16 5H17.5858L11.2929 11.2929C10.9024 11.6834 10.9024 12.3166 11.2929 12.7071C11.6834 13.0976 12.3166 13.0976 12.7071 12.7071L19 6.41421V8C19 8.55228 19.4477 9 20 9C20.5523 9 21 8.55228 21 8V4C21 3.44772 20.5523 3 20 3H16Z" fill="white"/>
```
Comment (codepantha) [inline index.html:150]: - [ ] Just like the previous reviewer suggested, there seems to be an overflow on the x-axis (left to right) on mobile devices as shown below 👇 Please fix this issue to achieve a more responsive design 📱 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUESTED ♻️ 

## Hi @MarsIncarnate 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ Used GitHub flow
✔️ No linter errors

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MarsIncarnate** [comment]
Hello reviewers @codepantha, @Tadesse-Alemayehu, could you help me with some pointers on how to resolve this overflow issue? I removed the 'overflow: hidden;' I previously set and my page is displayed perfectly on all screens but it breaks when I view the webpage using the responsive view in chrome inspect. I don't understand the reason for this issue and I can't find any helpful sources online, I couldn't do this project earlier this week because I got into an accident with my laptop. Please help me, I've already repeated a block I don't want to repeat this whole module.

**Comment by TadesseDev** [comment]
Give me a call on zoom @MarsIncarnate. Let's see it together.

File: index.js [hunk 5]
```diff
@@ -1,17 +1,111 @@
...
+];
+
+for (let i = 0; i < Details.length; i += 1) {
+  const item = portfolio.getElementsByClassName('portfolio-item')[i];
+  item.getElementsByTagName('img')[0].src = Details[i].img;
+  item.getElementsByTagName('ul')[0].innerHTML = Details[i].ul3.map((item) => `<li>${item}</li>`).join('');
+  item.getElementsByTagName('h3')[0].innerHTML = Details[i].h3;
+}
 
 function toggleMenu() {
```
Comment (huseyinbiyik) [inline index.js:75]: Kindly fix the overflowing issues on mobile as the previous reviewer requested. Please check the gif below.

Screen Recording 2023-01-07 at 11 55 09 PM

File: index.html [hunk 6]
```diff
@@ -91,82 +91,77 @@ <h2>My Recent Works
         <div class="portfolio-item">
```
Comment (huseyinbiyik) [inline index.html:91]: [OPTIONAL] As the main goal of this project kindly remove all the static content for in the middle and generate them dynamically with JavaScript. Check this [diff](https://github.com/olawale-o/olawale-o.me/pull/15/files#diff-0eb547304658805aad788d320f10bf1f292797b5e6d745a3bf617584da017051R95) to understand where you should remove.

**Review by huseyinbiyik — CHANGES_REQUESTED** [review]
Hi @MarsIncarnate,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
Almost there

## Highlights
- Linter checks passes 🟢

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**